From 96305388577d587324d01da4ecb86d8d13b44b85 Mon Sep 17 00:00:00 2001
From: Vaciller <wyc_19533480830@outlook.com>
Date: Sat, 11 Jul 2026 21:09:15 +0800
Subject: [PATCH 68/69] feat: add platform certificate, sharedUserId,
 TV_INPUT_HARDWARE permission, and prebuilt_etc for Rockchip SampleTvInput

Change-Id: Ib568424a9f5e86cf415715a272f7ae1333114c6b
---
 partner_support/samples/Android.bp          | 11 ++++++++++-
 partner_support/samples/AndroidManifest.xml | 17 ++++++++++-------
 2 files changed, 20 insertions(+), 8 deletions(-)

diff --git a/partner_support/samples/Android.bp b/partner_support/samples/Android.bp
index 4e71e061..1b6aa124 100644
--- a/partner_support/samples/Android.bp
+++ b/partner_support/samples/Android.bp
@@ -7,18 +7,21 @@ android_app {
 
     // Include all java files.
     srcs: ["src/**/*.java"],
+    certificate: "platform",
 
     static_libs: [
         "androidx.leanback_leanback",
         "androidx.tvprovider_tvprovider",
         "live-channels-partner-support",
     ],
+    required: ["com.example.partnersupportsampletvinput.xml"],
 
     optimize: {
         enabled: false,
     },
     // Overlay view related functionality requires system APIs.
-    sdk_version: "system_current",
+    //sdk_version: "system_current",
+    platform_apis: true,
     min_sdk_version: "23", // M
 
     // Required for com.android.tv.permission.RECEIVE_INPUT_EVENT
@@ -27,3 +30,9 @@ android_app {
     resource_dirs: ["res"],
 
 }
+
+prebuilt_etc {
+    name: "com.example.partnersupportsampletvinput.xml",
+    sub_dir: "permissions",
+    src: "com.example.partnersupportsampletvinput.xml",
+}
diff --git a/partner_support/samples/AndroidManifest.xml b/partner_support/samples/AndroidManifest.xml
index 08aa2317..9d307896 100644
--- a/partner_support/samples/AndroidManifest.xml
+++ b/partner_support/samples/AndroidManifest.xml
@@ -16,21 +16,24 @@
   -->
 
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-     xmlns:tools="http://schemas.android.com/tools"
-     package="com.example.partnersupportsampletvinput">
+    xmlns:tools="http://schemas.android.com/tools"
+    android:sharedUserId="android.uid.system"
+    package="com.example.partnersupportsampletvinput">
 
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
     <uses-permission android:name="android.permission.INTERNET"/>
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
     <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
+    <uses-permission android:name="android.permission.TV_INPUT_HARDWARE" />
     <!-- TODO: READ_EPG_DATA and WRITE_EPG_DATA need to be removed, once we fully
-                 migrate all test environment from LMP to MNC, because the permissions
+            migrate all test environment from LMP to MNC, because the permissions
                  are not required from MNC. -->
     <uses-permission android:name="com.android.providers.tv.permission.READ_EPG_DATA"/>
     <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA"/>
     <uses-permission android:name="com.android.tv.permission.RECEIVE_INPUT_EVENT"/>
-    <uses-sdk android:targetSdkVersion="29"
-         android:minSdkVersion="23"/>
+    <uses-permission android:name="android.permission.CAMERA" />
+    <uses-permission android:name="android.permission.RECORD_AUDIO" />
+    <uses-sdk android:targetSdkVersion="29" android:minSdkVersion="23"/>
     <!--TODO(b/68949299): remove tool hint when we have smaller dependency targets-->
     <application android:label="@string/partner_support_sample_tv_input"
          tools:replace="android:label,icon,theme,appComponentFactory"
@@ -45,8 +48,8 @@
             </intent-filter>
         </activity>
         <service android:name=".SampleTvInputService"
-             android:permission="android.permission.BIND_TV_INPUT"
-             android:label="@string/partner_support_sample_tv_input"
+            android:permission="android.permission.BIND_TV_INPUT"
+            android:label="@string/partner_support_sample_tv_input"
              android:process="com.example.partnersupportsampletvinput"
              android:exported="true">
             <intent-filter>
-- 
2.53.0

